oauth2_http/javatests/com/google/auth/oauth2/ImpersonatedCredentialsTest.java

```diff
@@ -142,6 +158,160 @@ private GoogleCredentials getSourceCredentials() throws IOException {
+    GoogleCredentials sourceCredentials = credentials.getSourceCredentials();
+    assertTrue(sourceCredentials instanceof UserCredentials);
+  }
+
+  @Test()
+  public void fromJson_userAsSource_WithoutQuotaProjectId() throws IOException {
+    GenericJson json =
+        buildImpersonationCredentialsJson(
+            IMPERSONATION_URL,
+            DELEGATES,
+            null,
+            USER_ACCOUNT_CLIENT_ID,
+            USER_ACCOUNT_CLIENT_SECRET,
+            REFRESH_TOKEN);
+    MockIAMCredentialsServiceTransportFactory mtransportFactory =
+        new MockIAMCredentialsServiceTransportFactory();
+    ImpersonatedCredentials credentials = ImpersonatedCredentials.fromJson(json, mtransportFactory);
+    assertEquals(IMPERSONATED_CLIENT_EMAIL, credentials.getAccount());
+    assertNull(credentials.getQuotaProjectId());
+    assertEquals(DELEGATES, credentials.getDelegates());
+    assertEquals(new ArrayList<String>(), credentials.getScopes());
+    assertEquals(credentials.getLifetime(), 3600);
+    GoogleCredentials sourceCredentials = credentials.getSourceCredentials();
+    assertTrue(sourceCredentials instanceof UserCredentials);
+  }
+
+  @Test()
+  public void fromJson_userAsSource_MissingDelegatesField() throws IOException {
+    GenericJson json =
+        buildImpersonationCredentialsJson(
+            IMPERSONATION_URL,
+            DELEGATES,
+            null,
+            USER_ACCOUNT_CLIENT_ID,
+            USER_ACCOUNT_CLIENT_SECRET,
+            REFRESH_TOKEN);
+    json.remove("delegates");
+    MockIAMCredentialsServiceTransportFactory mtransportFactory =
+        new MockIAMCredentialsServiceTransportFactory();
+    ImpersonatedCredentials credentials = ImpersonatedCredentials.fromJson(json, mtransportFactory);
+    assertEquals(IMPERSONATED_CLIENT_EMAIL, credentials.getAccount());
+    assertNull(credentials.getQuotaProjectId());
+    assertEquals(new ArrayList<String>(), credentials.getDelegates());
+    assertEquals(new ArrayList<String>(), credentials.getScopes());
+    assertEquals(credentials.getLifetime(), 3600);
+    GoogleCredentials sourceCredentials = credentials.getSourceCredentials();
+    assertTrue(sourceCredentials instanceof UserCredentials);
+  }
+
+  @Test()
+  public void fromJson_ServiceAccountAsSource() throws IOException {
+    GenericJson json =
+        buildImpersonationCredentialsJson(IMPERSONATION_URL, DELEGATES, QUOTA_PROJECT_ID);
+    MockIAMCredentialsServiceTransportFactory mtransportFactory =
+        new MockIAMCredentialsServiceTransportFactory();
+    ImpersonatedCredentials credentials = ImpersonatedCredentials.fromJson(json, mtransportFactory);
+    assertEquals(IMPERSONATED_CLIENT_EMAIL, credentials.getAccount());
+    assertEquals(QUOTA_PROJECT_ID, credentials.getQuotaProjectId());
+    assertEquals(DELEGATES, credentials.getDelegates());
+    assertEquals(new ArrayList<String>(), credentials.getScopes());
+    assertEquals(credentials.getLifetime(), 3600);
```

Comment:
expected value comes first; swap the arguments

Reply:
Done. I made the change in new commits to my branch https://github.com/liuchaoren/google-auth-library-java which should update the PR automatically. Somehow the new commits do not show up in the PR. There might be some lag. I will keep an eye on it.

Reply:
New commits show up in the PR now.